mern-ecommerce/controllers/user.js

```diff
@@ -40,6 +41,20 @@ exports.update = (req, res) => {
   );
 };
 
+exports.productById = (req, res, next, id) => { 
+  Product.findById(id) 
+    .populate('products.product', 'name price')
+    .exec((err, product) => { 
+      if (err || !product) {
+        return res.status(400).json({
+          error: errorHandler(err),
+        });
+      }
+      req.product = product; 
+      next();
+    });
```

Comment:
There is a typo in the last line of the `purchaseHistory` function. It should be `res.json(orders);` instead of `res.jsosn(orders);`.